lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -200,7 +207,195 @@ sub schedules {
+                                }
+                                if (!defined $machine_name) {
+                                    $machine_name = $yaml->{$group}->{defaults}->{$arch}{machine};
+                                }
+                                my $machine      = $machines->find({name => $machine_name});
+                                my $product_spec = $yaml->{$group}->{products}{$product_name};
+                                my $product      = $products->find(
+                                    {
+                                        arch    => $arch,
+                                        distri  => $product_spec->{distribution},
+                                        flavor  => $product_spec->{flavor},
+                                        version => $product_spec->{version},
+                                    });
+                                die "Machine '$machine_name' is invalid\n" unless $machine;
+                                die "Product '$product_name' is invalid\n" unless $product;
+
+                                my $test_suite = $test_suites->find_or_create({name => $testsuite_name});
+                                my $r          = $job_templates->find_or_create(
+                                    {
+                                        product_id    => $product->id,
+                                        machine_id    => $machine->id,
+                                        group_id      => $id,
+                                        test_suite_id => $test_suite->id,
+                                    });
+
+                                $r->product_id($product->id);
+                                $r->machine_id($machine->id);
+                                if ($prio) {
+                                    $r->prio($prio);
+                                }
+                                $r->update;
+
+                                # Stop iterating if there were errors with this test suite
+                                last if (@$errors);
+                            }
+                        }
+                    }
+
+                    # Preview mode: Get the expected YAML and rollback the result
+                    if ($self->param('preview')) {
+                        $json->{template} = $self->get_job_groups($json->{id});
+                        $self->schema->txn_rollback;
+                    }
+                });
+        }
+        catch {
+            # Push the exception to the list of errors without the trailing new line
+            push @$errors, substr($_, 0, -1);
+        };
+
+        if (@$errors) {
+            $json->{error} = \@$errors;
+            $self->app->log->error(@$errors);
+            $self->respond_to(json => {json => $json, status => 400},);
+            return;
+        }
+
+        $self->emit_event('openqa_jobtemplate_create', $json);
+        $self->respond_to(json => {json => $json});
```

Comment:
It would be nice if it would respond YAML, at least when requested by the client - at least when no error occurred. Then on the JavaScript-side the result would also be shown as YAML.